### PR TITLE
[ODLA] odla_popart exception improvement

### PR DESCRIPTION
**Pull request by youbeny**
odla_popart exception improvement

### CODE REVIEW

File: ODLA/include/ODLA/odla_common.h [hunk 1]
```diff
@@ -135,6 +135,13 @@ typedef void odla_void;
...
 //! \brief Return status
 typedef enum {
+  ODLA_DL_ERROR,
+  ODLA_FILE_NOT_EXIST,
+  ODLA_INVALID_PARAM,
+  ODLA_MEM_ERROR,
+  ODLA_NULLPTR,
+  ODLA_UNSUPPORTED_DATATYPE,
+  ODLA_UNSUPPORTED_DEVICE_TYPE,
   ODLA_SUCCESS,
```
Comment (weimingzha0) [inline ODLA/include/ODLA/odla_common.h:145]: Add the new enums after ODLA_SUCCESS. 
So success will always == 0.

File: ODLA/include/ODLA/odla_common.h [hunk 2]
```diff
@@ -135,6 +135,13 @@ typedef void odla_void;
 
 //! \brief Return status
 typedef enum {
+  ODLA_DL_ERROR,
```
Comment (weimingzha0) [inline ODLA/include/ODLA/odla_common.h:138]: What is DL_ERROR ?
Add some doxygen comments for each error?

File: ODLA/include/ODLA/odla_common.h [hunk 3]
```diff
@@ -135,6 +135,13 @@ typedef void odla_void;
...
+  ODLA_FILE_NOT_EXIST,
+  ODLA_INVALID_PARAM,
+  ODLA_MEM_ERROR,
+  ODLA_NULLPTR,
```
Comment (weimingzha0) [inline ODLA/include/ODLA/odla_common.h:142]: ODLA_NULLPTR is same as ODLA_INVALID_PARAM ?

File: ODLA/include/ODLA/odla_common.h (hunk 2 above)
Comment (weimingzha0) [reply ODLA/include/ODLA/odla_common.h:138]: maybe ODLA_UNSUPPORTED_OP?

Comment (youbeny) [reply ODLA/include/ODLA/odla_common.h:138]: DL_ERROR  means encounter error(s) when dlopen a shared library

File: ODLA/include/ODLA/odla_common.h (hunk 1 above)
Comment (youbeny) [reply ODLA/include/ODLA/odla_common.h:145]: agree

File: ODLA/include/ODLA/odla_common.h (hunk 3 above)
Comment (youbeny) [reply ODLA/include/ODLA/odla_common.h:142]: ok

**Review by weimingzha0 — APPROVED** [review]

**Pull request merged**